File: youpinhui-sellergoods-interface/src/main/java/com/youpinhui/sellergoods/service/TypeTemplateService.java
package com.youpinhui.sellergoods.service;
import java.util.List;
import java.util.Map;

import com.youpinhui.entity.PageResult;
import com.youpinhui.pojo.TbTypeTemplate;


/**
 * provider TypeTemplate interface
 * @author Leon
 *
 */
public interface TypeTemplateService {

	/**
	 * find all
	 * @return
	 */
	public List<TbTypeTemplate> findAll();
	
	
	/**
	 * query according pagination 
	 * @return
	 */
	public PageResult findPage(int pageNum,int pageSize);
	
	
	/**
	 * Add
	*/
	public void add(TbTypeTemplate typeTemplate);
	
	
	/**
	 * update
	 */
	public void update(TbTypeTemplate typeTemplate);
	

	/**
	 * get one by id
	 * @param id
	 * @return
	 */
	public TbTypeTemplate findOne(Long id);
	
	
	/**
	 * batch delete
	 * @param ids
	 */
	public void delete(Long [] ids);

	/**
	 * fuzzy search 
	 * if the object  is not null , get the parameter from it as the search condition.
	 * @param pageNum current page number
	 * @param pageSize size of the page to display
	 * @return
	 */
	public PageResult findPage(TbTypeTemplate typeTemplate, int pageNum,int pageSize);


	public List<Map> selectOptionList();
	
	
	//get the specification list by the type_template id;
	public List<Map> findSpecList(Long id);
	
}

File: youpinhui-shop-web/src/main/webapp/js/service/typeTemplateService.js
//Sevice
app.service('typeTemplateService',function($http){
	    	
	 //test case , get all the data return a Json
	this.findAll=function(){
		return $http.get('../typeTemplate/findAll.do');		
	}
	//pagination
	this.findPage=function(page,rows){
		return $http.get('../typeTemplate/findPage.do?page='+page+'&rows='+rows);
	}
	//find one object
	this.findOne=function(id){
		return $http.get('../typeTemplate/findOne.do?id='+id);
	}
	//add 
	this.add=function(entity){
		return  $http.post('../typeTemplate/add.do',entity );
	}
	//update 
	this.update=function(entity){
		return  $http.post('../typeTemplate/update.do',entity );
	}
	//batch delete
	this.dele=function(ids){
		return $http.get('../typeTemplate/delete.do?ids='+ids);
	}
	//search
	this.search=function(page,rows,searchEntity){
		return $http.post('../typeTemplate/search.do?page='+page+"&rows="+rows, searchEntity);
	}
	
	this.selectOptionList=function(){
		return $http.get('../typeTemplate/selectOptionList.do');
	}
	this.findSpecList=function(id){
		return $http.get('../typeTemplate/findSpecList.do?id='+id);
	}

});
